miio/vacuumcontainers.py

```diff
@@ -197,8 +197,9 @@ def __init__(self, data: Union[List[Any], Dict[str, Any]]) -> None:
                 "clean_time": data[0],
                 "clean_area": data[1],
                 "clean_count": data[2],
-                "records": data[3],
             }
+            if len(data) > 3:
+                self.data["records"] = data[3]
```

Comment:
```suggestion
            self.data["records"] = data[3] if len(data) > 3 else []
```
This would also allow removing the records check below and making the getter simpler. Otherwise this PR looks good to me.

Reply:
Not wanting to discuss styling decisions but the ternary operator is not very pythonic in my opinion. 

And in what way does it make the record check below obsolete? We still need to catch the case in which the new dict-style parameter has no records coming from the device. Maybe it would be more concise to move the check from the getter up here.

Reply:
Ah, sorry, you're right. I was not thinking about the dict input, just about keeping the getter clean... I'd say it's your call, and I agree with the notion that ternaries are less readable than proper if-elses.